### DISCUSSION TITLE
Custom style manager

### DISCUSSION THREAD

**Discussion by Nelida27** [GrapesJS/grapesjs#4028] [General]
Hi there,
I want to change style manager properties based on selected component.
I have seen that it is implemented when component is type of body.
Any idea how to customize it even for other components?

Thanks.

**Comment by ronaldohoch** [comment]
You can create a new component and set the stylables property inside each one: https://github.com/artf/grapesjs-mjml/blob/master/src/components/Text.js#L26

Here some links to help: https://grapesjs.com/docs/modules/Components.html#how-components-work
https://grapesjs.com/docs/modules/Components.html

**Reply by Nelida27** [reply]
Thank u Ronaldo for your response.
But I do not want to create another component, I just want to change Style Manager based on selected components.
ex: If I have selected text component Style Manager will show only Basics, if i select link component I want to see Basic and Margin into Style Manger.

**Reply by ronaldohoch** [reply]
Well...

It will be worst and big, but i can figure another way.:

- Listen to event `component:selected` 
- check which component are selected by checking the model tagname or other property
- Clear the style manager panel
- - You can use editor.StyleManager.getSectors() to get all sectors and loop it to remove with removeSector
- - - https://grapesjs.com/docs/api/style_manager.html#stylemanager
- Set the styles by the component using setSector
- - https://grapesjs.com/docs/api/style_manager.html#addsector
- - And add propertys
- - - https://grapesjs.com/docs/api/style_manager.html#addproperty

Never made it but i think it work.
